pkg/querier/queryrange/shard_resolver.go

```diff
@@ -265,37 +269,43 @@ func (r *dynamicShardResolver) ShardingRanges(expr syntax.Expr, targetBytesPerSh
 		if resp, ok := httpgrpc.HTTPResponseFromError(err); ok && (resp.Code == http.StatusNotFound) {
```

Comment:
We still need this fallback for ingesters, right?

Reply:
I should probably remove this code for consistency with removing the same in `gatewayclient`